backend/lcfs/web/api/compliance_report/update_service.py

```diff
@@ -138,34 +169,57 @@ async def handle_submitted_status(self, report: ComplianceReport):
         # Save the summary
         if report.summary:
             # Update existing summary
-            await self.repo.save_compliance_report_summary(
+            report.summary = await self.repo.save_compliance_report_summary(
                 report.compliance_report_id, calculated_summary
             )
         else:
             # Create new summary if it doesn't exist
             new_summary = await self.repo.add_compliance_report_summary(
                 calculated_summary
             )
-            report.summary = new_summary
             # Update the report with the new summary
-            await self.repo.update_compliance_report(report)
+            report.summary = new_summary
+        # Create a new reserved transaction for receiving organization
+        report.transaction = await self.org_service.adjust_balance(
+            transaction_action=TransactionActionEnum.Reserved,
+            compliance_units=report.summary.line_20_surplus_deficit_units,
+            organization_id=report.organization_id,
+        )
+        await self.repo.update_compliance_report(report)
 
         return calculated_summary
 
     async def handle_recommended_by_analyst_status(self, report: ComplianceReport):
         """Handle actions when a report is Recommended by analyst."""
         # Implement logic for Recommended by analyst status
-        pass
+        user = self.request.user
+        has_director_role = user_has_roles(
```

Comment:
Please update these role checks with a variable like "has_roles" instead so we don't confuse the director value.

Reply:
updated